addons/portal/controllers/portal.py

```diff
@@ -350,9 +349,8 @@ def attachment_add(self, name, file, thread_model, thread_id, access_token=None)
             'res_id': 0,
             'access_token': IrAttachment._generate_access_token(),
         })
-        return request.make_response(
-            data=json.dumps(attachment.read(['id', 'name', 'mimetype', 'file_size', 'access_token'])[0]),
-            headers=[('Content-Type', 'application/json')]
+        return request.make_json_response(
```

Comment:
I replaced the uploader root of the portal chatter with this one. 
It's not possible for now to delete this controller completely because rating portal composer still uses it. Could be done in the follow up PR to remove the composer widget as well.